src/enhanced/utils/queue/index.ts

```diff
@@ -0,0 +1,51 @@
+import { SQS } from '@aws-sdk/client-sqs';
+import config from 'config';
+import { JobsQueuePayload } from '../../types/job-types';
+import { isJobQueuePayload } from '../../types/utils/type-guards';
+
+/**
+ * This function listens to the job queue until a message is received.
+ * @returns {Promise<JobsQueuePayload>} the promise for the payload object after a message has been received
+ */
+export async function listenToJobQueue(): Promise<JobsQueuePayload> {
+  const region = config.get<string>('aws_region');
+  const queueUrl = config.get<string>('jobsQueueUrl');
+
+  const client = new SQS({ region });
+
+  // We want to loop indefinitely so that we continue to poll the queue.
+  while (true) {
+    const receiveMessage = { QueueUrl: queueUrl, MaxNumberOfMessages: 1, WaitTimeSeconds: 2 };
+
+    const res = await client.receiveMessage(receiveMessage);
+
+    if (!res.Messages || res.Messages.length === 0) continue;
+
+    const message = res.Messages[0];
+
+    // We have the message body, now we can delete it from the queue.
+    client.deleteMessage({ QueueUrl: queueUrl, ReceiptHandle: message.ReceiptHandle });
```

Comment:
this returns a promise. would we want to await for this. and as for any awaits, should wrap in try/catch block to catch any client failures. besides this LGTM! thanks for adding the message delete!

Reply:
Ah, thank you for pointing this out. Rookie mistake! lol